core/src/main/java/org/ehcache/EhcacheManager.java

```diff
@@ -264,6 +264,10 @@ private boolean isDiskTransient(Ehcache<?, ?> ehcache) {
     if (config.getResourcePools().getResourceTypeSet().contains(ResourceType.Core.DISK)) {
       LocalPersistenceService persistenceService = serviceLocator.getService(LocalPersistenceService.class);
       
+      if (persistenceService == null) {
+        throw new IllegalStateException("No LocalPersistenceService could be found - did you configure it at the CacheManager level?");
+      }
+
```

Comment:
I don't think we want our builders to have any kind of intelligence w.r.t the logical validity of what they're building.